File: src/leetcode/2026/01/Solution520.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution520 {
public:
    bool detectCapitalUse(string word) {
        int n = word.size();
        if (n == 0) return true;

        bool isFirstUpper = isupper(word[0]);
        bool isSecondUpper = n > 1 ? isupper(word[1]) : false;

        for (int i = 1; i < n; ++i) {
            if (isSecondUpper) {
                if (!isFirstUpper || !isupper(word[i])) {
                    return false;
                }
            } else {
                if (isupper(word[i])) {
                    return false;
                }
            }
        }

        return true;
    }
};

int main() {
    Solution520 solution;

    string word1 = "USA";
    cout << (solution.detectCapitalUse(word1) ? "true" : "false") << endl; // Expected output: true

    string word2 = "FlaG";
    cout << (solution.detectCapitalUse(word2) ? "true" : "false") << endl; // Expected output: false

    string word3 = "mL";
    cout << (solution.detectCapitalUse(word3) ? "true" : "false") << endl; // Expected output: false

    return 0;
}

File: src/leetcode/2026/01/Solution520.java
public class Solution520 {
    public boolean detectCapitalUse(String word) {
        int n = word.length();
        if (n == 0) {
            return true;
        }

        boolean isFirstUpper = Character.isUpperCase(word.charAt(0));
        boolean isSecondUpper = n > 1 && Character.isUpperCase(word.charAt(1));

        for (int i = 1; i < n; i++) {
            boolean isCurrentUpper = Character.isUpperCase(word.charAt(i));
            if (isSecondUpper) {
                if (!isFirstUpper || !isCurrentUpper) {
                    return false;
                }
            } else {
                if (isCurrentUpper) {
                    return false;
                }
            }
        }

        return true;
    }

    public static void main(String[] args) {
        Solution520 solution = new Solution520();

        // String word1 = "USA";
        // System.out.println("Is capital use correct for '" + word1 + "'? " +
        // solution.detectCapitalUse(word1));

        // String word2 = "FlaG";
        // System.out.println("Is capital use correct for '" + word2 + "'? " +
        // solution.detectCapitalUse(word2));

        String word3 = "mL";
        System.out.println("Is capital use correct for '" + word3 + "'? " + solution.detectCapitalUse(word3));
    }
}
